docs/src/koopman/dmd.md

```diff
@@ -0,0 +1,231 @@
+## Dynamic Mode Decomposition
+
+
+#### (Exact) Dynamic Mode Decomposition (DMD)
+
+Dynamic Mode Decomposition, or Exact Dynamic Mode Decomposition, is a method for
+generating an approximating linear differential equation (the Koopman operator)
+from the observed data.
+
+To construct the approximation, use:
+
+```julia
+ExactDMD(X; dt = 0.0)
+```
+
+-  `X` is a data matrix of the observations over time, where each column is
+  all observables at a given time point.
+- `dt` is an optional value for the spacings
+  between the observation timepoints which is used in the construction of the
+  continuous dynamics.
+
+#### Extended Dynamic Mode Decomposition (eDMD)
+
+Extended Dynamic Mode Decomposition (eDMD), is a method for
+generating an approximating linear differential equation (the Koopman operator)
+for the observed data in a chosen basis of observables. Thus the signature is:
+
+```julia
+ExtendedDMD(X,basis; dt = 1.0)
+```
+
+-  `X` is a data matrix of the observations over time, where each column is
+  all observables at a given time point.
+- `dt` is an optional value for the spacings
+  between the observation timepoints which is used in the construction of the
+  continuous dynamics.
+
+#### Shared DMD Features
+
+To get the dynamics from the DMD object, use the `dynamics` function:
+
+```julia
+dynamics(dmd, discrete=true)
+```
+
+This will build the `f` function for DifferentialEquations.jl integrators, and
+defaults to building approximations for `DiscreteProblem`, but this is changed
+to approximating `ODEProblem`s by setting `discrete=false`.
+
+The linear approximation can also be analyzed using the following functions:
+
+```julia
+eigen(dmd)
+eigvals(dmd)
+eigvecs(dmd)
+```
+
+Additionally, DMD objects can be updated to add new measurements via:
+
+```julia
+update!(dmd, x; Δt = 0.0, threshold = 1e-3)
+```
+
+## Examples
+
+### Structural Identification with SInDy
```

Comment:
these should probably move out to tutorials pages which come before the reference pages.